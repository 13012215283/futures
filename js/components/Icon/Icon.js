import React from 'react';
import PropTypes from 'prop-types';
import { Image, View, ViewPropTypes } from 'react-native';
import { px2dp } from '../../utils/px2dp';

export default function Icon({ imgUrl, containerStyle, imgStyle }) {
  return (
    <View style={containerStyle}>
      {imgUrl !== '' && <Image source={{ uri: imgUrl }} style={imgStyle} />}
    </View>
  );
}

Icon.propTypes = {
  imgUrl: PropTypes.string,
  /* eslint-disable react/no-typos */
  containerStyle: ViewPropTypes.style,
  imgStyle: Image.propTypes.style,
};

Icon.defaultProps = {
  imgUrl: '',
  containerStyle: {
    width: '100%',
    height: px2dp(140),
    borderRadius: px2dp(8),
    marginTop: px2dp(32),
    justifyContent: 'center',
    alignItems: 'center',
  },
  imgStyle: {
    width: px2dp(140),
    height: px2dp(140),
  },
};
