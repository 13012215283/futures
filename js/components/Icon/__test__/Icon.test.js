import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../Icon';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Icon imgUrl="https://ss1.bdstatic.com/70cFuXSh_Q1YnxGkpoWK1HF6hhy/it/u=49366202,632101467&fm=27&gp=0.jpg" />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly without set any child', () => {
  const tree = renderer.create(<Icon />).toJSON();

  expect(tree).toMatchSnapshot();
});
