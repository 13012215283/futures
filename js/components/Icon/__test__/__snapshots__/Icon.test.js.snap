// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 8,
      "height": 140,
      "justifyContent": "center",
      "marginTop": 32,
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://ss1.bdstatic.com/70cFuXSh_Q1YnxGkpoWK1HF6hhy/it/u=49366202,632101467&fm=27&gp=0.jpg",
      }
    }
    style={
      Object {
        "height": 140,
        "width": 140,
      }
    }
  />
</View>
`;

exports[`renders correctly without set any child 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 8,
      "height": 140,
      "justifyContent": "center",
      "marginTop": 32,
      "width": "100%",
    }
  }
/>
`;
